lib/public/views/Tags/Details/TagDetailsModel.js

```diff
@@ -0,0 +1,253 @@
+import { getRemoteData } from '../../../utilities/fetch/getRemoteData.js';
+import { jsonFetch } from '../../../utilities/fetch/jsonFetch.js';
+import { Observable, RemoteData } from '/js/src/index.js';
+
+export const TAG_DETAILS_PANELS_KEYS = {
+    LOGS: 'logs',
+};
+
+/**
+ * Model storing a given tag details state
+ */
+export class TagDetailsModel extends Observable {
+    /**
+     * Constructor
+     *
+     * @param {number|null} tagId the id of the tag this model represents
+     * @param {string} panelKey the key of the panel to display
+     */
+    constructor(tagId = null, panelKey) {
+        super();
+
+        this._tagDetails = RemoteData.notAsked();
+        this._currentPanelData = RemoteData.notAsked();
+
+        this.tagId = tagId;
+        this.currentPanelKey = panelKey;
+
+        this._tagChanges = {};
+        this._isEditModeEnabled = false;
+    }
+
+    /**
+     * Returns the remote data representing the currently displayed tag
+     * @return {RemoteData} the tag's remote data
+     */
+    get tagDetails() {
+        return this._tagDetails;
+    }
+
+    /**
+     * Defines the current log by its id
```

Comment:
This file needs another close look to all doc blocks as I have identified multiple missmatches